odbc/src/api/diag.rs

```diff
@@ -40,34 +25,6 @@ pub unsafe fn set_sql_statew(sql_state: &str, output_ptr: *mut WideChar) {
     copy_nonoverlapping(state_u16.as_ptr(), output_ptr, 6);
 }
 
-///
-/// get_diag_rec copies the given ODBC error's diagnostic information
-/// into the provided pointers.
-///
-/// # Safety
-/// This writes to multiple raw C-pointers
-///
-pub unsafe fn get_diag_rec(
```

Comment:
This looks good! I think we can simplify the `get_diag_field` function now as well to remove the `is_wstring` flag